src/linter/yaml/YamlLinter.ts

```diff
@@ -0,0 +1,88 @@
+import {LintMessageSeverity} from "../LinterContext.js";
+import LinterContext from "../LinterContext.js";
+import deprecatedLibraries from "../../utils/deprecatedLibs.js";
+import {DataWithPosition, fromYaml, getPosition} from "data-with-position";
+
+interface YamlContent extends DataWithPosition {
+	framework?: {
+		libraries?: {
+			name: string;
+		}[];
+	};
+	positionKey?: {
+		end: {
+			column: number;
+			line: number;
+		};
+		start: {
+			column: number;
+			line: number;
+		};
+	};
+}
+
+export default class YamlLinter {
+	#content;
+	#resourcePath;
+	#context: LinterContext;
+
+	constructor(resourcePath: string, content: string, context: LinterContext) {
+		this.#content = content;
+		this.#resourcePath = resourcePath;
+		this.#context = context;
+	}
+
+	// eslint-disable-next-line @typescript-eslint/require-await
+	async lint() {
+		try {
+			/* Support multiple documents in one Yaml file
+			https://sap.github.io/ui5-tooling/stable/pages/extensibility/CustomTasks/#example-custom-task-extension-defined-in-ui5-project */
+
+			// Split Yaml file into part documents by '---' separator
+			const partDocuments = this.#content.split(/(\r?\n|\r|\n)---/g).map((part) => part.trim());
```

Comment:
I thought `data-with-position` is capable of handling multiple documents in one stream, but it seems it isn't, right?

One problem I noticed now is that the line/column information for the second document (probably all except the first) is not correct. At least in this example I get findings for line 3 (correct) and 6 (should be 9):
```yaml
framework:
  libraries:
    - name: sap.ui.commons

---

framework:
  libraries:
    - name: sap.ui.commons
```

Reply:
Yes, looks good to me 👍🏻